### PR TITLE
PhpUnit refactoring - Test driven development

### PR DESCRIPTION
**Pull request by l-lemaitre**

### CODE REVIEW

File: src/Controller/TaskController.php [hunk 1]
```diff
@@ -7,18 +7,20 @@
...
+    #[Route('/tasks', name: 'app_task_list')]
+    public function listAction(ManagerRegistry $doctrine): Response
     {
         return $this->render('task/list.html.twig', ['tasks' => $doctrine->getRepository(Task::class)->findAll()]);
     }
 
-    #[Route('/tasks/create', name: 'task_create')]
-    public function createAction(ManagerRegistry $doctrine, Request $request)
+    #[Route('/tasks/create', name: 'app_task_create')]
+    public function createAction(ManagerRegistry $doctrine, Request $request): RedirectResponse|Response
```
Comment (IcheNouri) [inline src/Controller/TaskController.php]: injecter ManagerRegistry une seule fois dans le constructor

File: src/Controller/TaskController.php [hunk 2]
```diff
@@ -62,30 +62,26 @@ public function editAction(Task $task, Request $request)
...
 
-    /**
-     * @Route("/tasks/{id}/delete", name="task_delete")
-     */
-    public function deleteTaskAction(Task $task)
+    #[Route('/tasks/{id}/delete', name: 'app_task_delete')]
+    public function deleteTaskAction(ManagerRegistry $doctrine, Task $task): RedirectResponse
     {
-        $em = $this->getDoctrine()->getManager();
+        $em = $doctrine->getManager();
```
Comment (IcheNouri) [inline src/Controller/TaskController.php]: tu peux la variable $em (entity manager) en variable de class, après tu peux faire $this->em;

File: src/Controller/UserController.php [hunk 3]
```diff
@@ -4,64 +4,88 @@
...
+            // encode the plain password
+            $user->setPassword(
+                $userPasswordHasher->hashPassword(
+                    $user,
+                    $user->getPassword()
+                )
+            );
 
+            $em = $doctrine->getManager();
             $em->persist($user);
```
Comment (IcheNouri) [inline src/Controller/UserController.php]: tester avec et sans persist et voir les recommendation de symfony

File: tests/Controller/UserControllerTest.php [hunk 4]
```diff
@@ -246,35 +254,64 @@ public function testEditActionBeyondMaxFields()
...
+        $user->setPassword($userData['password']);
+        $user->setEmail($userData['email']);
+        $this->entityManager->persist($user);
+        $this->entityManager->flush();
+
+        $this->client->loginUser($this->user);
+
+        $urlGenerator = $this->client->getContainer()->get('router');
+
+        $this->client->request(Request::METHOD_GET, $urlGenerator->generate('app_user_delete', ['id' => 2]));
```
Comment (IcheNouri) [inline tests/Controller/UserControllerTest.php]: à tester si tu peux remplacer 2 par $user->getId()

File: src/Controller/TaskController.php (hunk 1 above)
Comment (l-lemaitre) [reply src/Controller/TaskController.php]: Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)

File: src/Controller/TaskController.php (hunk 2 above)
Comment (l-lemaitre) [reply src/Controller/TaskController.php]: Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)

File: src/Controller/UserController.php (hunk 3 above)
Comment (l-lemaitre) [reply src/Controller/UserController.php]: Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)

File: tests/Controller/UserControllerTest.php (hunk 4 above)
Comment (l-lemaitre) [reply tests/Controller/UserControllerTest.php]: Fixed => See commit [Issue #4 Implementation of automated tests](https://github.com/l-lemaitre/todolist/pull/9/commits/cddb13e1ca46e6ffbaba7587e68c926c7aedbaa7)

**Pull request merged**